File: src/others/findDifferentElements/findDifferentElements.go
package finddifferentelements

func findDifferentElements(arrayA, arrayB []int) []int {
	mapA := make(map[int]int)
	mapB := make(map[int]int)

	for i, ele := range arrayA {
		mapA[ele] = i
	}

	for i, ele := range arrayB {
		mapB[ele] = i
	}

	for key := range mapA {
		if _, isExist := mapB[key]; isExist {
			delete(mapA, key)
			delete(mapB, key)
		}
	}

	var results []int

	for key := range mapA {
		results = append(results, key)
	}

	for key := range mapB {
		results = append(results, key)
	}

	return results
}

File: src/others/findDifferentElements/findDifferentElements.ts
/**
 * @template T
 * @param {Array<T>} arrayA
 * @param {Array<T>} arrayB
 * @returns {Array<T>}
 */
function findDifferentElements<T>(arrayA: Array<T>, arrayB: Array<T>): Array<T> {
  const mapA: Map<T, undefined> = new Map();
  const mapB: Map<T, undefined> = new Map();

  for (const ele of arrayA) {
    mapA.set(ele, undefined);
  }

  for (const ele of arrayB) {
    mapB.set(ele, undefined);
  }

  for (const key of mapA.keys()) {
    if (mapB.has(key)) {
      mapA.delete(key);
      mapB.delete(key);
    }
  }

  return [...mapA.keys(), ...mapB.keys()];
}

interface Params {
  [key: string]: object;
  input: {
    [key: string]: string | (() => void);
    name: string;
    onChange: () => void;
    value: string;
  };
  meta: object;

}
